### DISCUSSION TITLE
Is overwriting PermissionFilter possible?

### DISCUSSION THREAD

**Discussion by yassinedoghri** [codeigniter4/shield#763] [Q&A] [answered]
Is it possible to overwrite the `PermissionFilter`? If so, how should it be done?

I see it's defined in `src/Config/Registrar.php` but when I try overwriting it using my own Registrar under `app/Config` for example, it just does not seem to do anything.

Apparently Shield's Registrar gets loaded last, thus overwriting anything set in the `app` folder.

Thank you for this great library by the way! And thanks in advance for any help.

**Comment by kenjis** [comment, marked as answer]
[Registrar](https://codeigniter4.github.io/CodeIgniter4/general/configuration.html#registrars) is a feature to add or overwrite `app/Config/*` settings.
There is no way to disable a specific Registrar.

1. You can overwrite it after Registrar works.

```diff
--- a/app/Config/Filters.php
+++ b/app/Config/Filters.php
@@ -61,4 +61,11 @@ class Filters extends BaseConfig
      * 'isLoggedIn' => ['before' => ['account/*', 'profiles/*']]
      */
     public array $filters = [];
+
+    public function __construct()
+    {
+        parent::__construct(); // Registrar works in this constructor.
+
+        $this->aliases['permission'] = 'App\Filters\PermissionFilter';
+    }
 }
```

2. You can [disable Auto-discovery of Registrars](https://codeigniter4.github.io/CodeIgniter4/general/modules.html#specify-discovery-items). 
   But if you disable it, you need to add all config items that all packages provide with Registrars to your `app/Config/*`.

**Reply by yassinedoghri** [reply]
Thank you for the insight! I finally understand what Registrars are for.

I went with the first option by overwriting in the filter's constructor and it works great 👍
